### PR TITLE
[MM-37687] Add pagination to channel list command

### PR DESCRIPTION
**Pull request by mgdelacroix**
#### Summary
This PR adds a loop for each of the channel fetch functions, so the `channel list` command doesn't hit the max per page limit in the server and effectively fetches all the channels available.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-37687

### CODE REVIEW

**Review by lindy65 — APPROVED** [review]
Thanks @mgdelacroix - this will be tested after merge 👍

File: commands/channel.go [hunk 1]
```diff
@@ -263,6 +265,48 @@ func archiveChannelsCmdF(c client.Client, cmd *cobra.Command, args []string) err
...
+	channels := []*model.Channel{}
+	page := 0
+
+	for {
+		channelsPage, response := c.GetPublicChannelsForTeam(teamID, page, web.PerPageMaximum, "")
+		if response.Error != nil {
+			return nil, response.Error
+		}
+
+		if len(channelsPage) == 0 {
```
Comment (larkox) [inline commands/channel.go:278]: Probably not needed, but should we check later also `len(channelsPage) < web.PerPageMaximum` to save us from one more call?

**Review by larkox — APPROVED** [review]
LGTM. Just two small comments.

File: commands/channel.go [hunk 2]
```diff
@@ -271,25 +315,25 @@ func listChannelsCmdF(c client.Client, cmd *cobra.Command, args []string) error
 			continue
 		}
 
-		publicChannels, response := c.GetPublicChannelsForTeam(team.Id, 0, 10000, "")
-		if response.Error != nil {
-			printer.PrintError("Unable to list public channels for '" + args[i] + "'. Error: " + response.Error.Error())
+		publicChannels, err := getAllPublicChannelsForTeam(c, team.Id)
+		if err != nil {
+			printer.PrintError(fmt.Sprintf("unable to list public channels for %q: %s", args[i], err))
```
[the remark this answers is not in the document]
Comment (mgdelacroix) [reply commands/channel.go:320]: The change was intended, just changing for consistency

File: commands/channel.go (hunk 1 above)
Comment (mgdelacroix) [reply commands/channel.go:278]: We've found instances in which although the page comes with less than the total amount of elements, there are more pages to fetch (for example, when the items are indexed on ES/bleve) hence we stop only when the page fetched comes empty, to be 100% sure that there are no more elements to fetch

**Review by isacikgoz — APPROVED** [review]
LGTM 👍

**Pull request merged**